src/Propulsion.SqlStreamStore/SqlStreamStoreSource.fs

```diff
@@ -0,0 +1,71 @@
+namespace Propulsion.SqlStreamStore
+
+open System
+open SqlStreamStore
+
+type SqlStreamStoreSource =
+
+    static member internal RunInternal
+            (logger : Serilog.ILogger,
+             store: IStreamStore,
+             checkpointer: ICheckpointer,
+             sink : Propulsion.ProjectorPipeline<_>,
+             consumerGroup: string,
+             ?statsInterval: TimeSpan,
+             ?readerMaxBatchSize: int,
+             ?readerSleepInterval: TimeSpan,
+             ?readerStopAfterInterval: TimeSpan) : Async<unit> =
+
+        async {
+            let streamId = "$all"
+
+            let logger =
+                let instanceId = System.Guid.NewGuid()
+                logger
+                    .ForContext("instanceId", string instanceId)
+                    .ForContext("consumerGroup", consumerGroup)
+
+            let ingester : Propulsion.Ingestion.Ingester<_,_> =
+                sink.StartIngester(logger, 0)
+
+            let reader =
+                StreamReader(logger,
+                             store,
+                             checkpointer,
+                             ingester.Submit,
+                             streamId,
+                             consumerGroup,
+                             ?maxBatchSize = readerMaxBatchSize,
+                             ?sleepInterval = readerSleepInterval,
+                             ?statsInterval = statsInterval,
+                             ?stopAfterInterval = readerStopAfterInterval)
+
+            try
+                let! position =
+                    checkpointer.GetPosition { Stream = streamId; ConsumerGroup = consumerGroup }
```

Comment:
replace record with an argument list?

Reply:
I went for a record to keep it in line with the other interface member, if I'm changing it, I'll change both members.

Reply:
👍 In general, using records in this way is not the convention in Propulsion